### PR TITLE
Refactor Message

### PR DESCRIPTION
**Pull request by Jarema**
This is first pass of changes in messages.

As I was going back and forth with how to restrict available methods on `BoundMessage` depending on context (JetStream or not) I decided to first include `BoundMessage` with POD `Message` struct.

Next step is to restrict the methods.

### CODE REVIEW

**Comment by Jarema** [comment]
solves #227

**Review by caspervonb — COMMENTED** [review]
Seeing it, the echo of `message.message` doesn't feel great 🤔 
Its only like 3 basic fields, so maybe we should just embed it and provide into/as conversions?

**Comment by Jarema** [comment]
We agreed to embed fields of Message into BoundMessage, so we dont have echo and provide methods for transforming raw Message into BoundMessage (for internal usage) and BoundMessage into Message for Users.

That allows users to create Message and pass it to client while removing echo message.message and NOT working around it with Deref.
It also allows users to transform BoundMessage from client into Message if needed.

File: src/asynk.rs [hunk 1]
```diff
@@ -92,12 +92,15 @@ use std::fmt;
...
 use std::net::IpAddr;
 use std::path::Path;
+use std::sync::atomic::Ordering;
 use std::sync::{atomic::AtomicBool, Arc};
 use std::time::Duration;
 
 use blocking::unblock;
 use crossbeam_channel::{Receiver, Sender};
 
+use crate::message::IntoBound;
```
Comment (caspervonb) [inline src/asynk.rs:102]: Hmm wonder if we need this `IntoBound` trait at all? Why would an end-user do this conversion?

File: src/message.rs [hunk 2]
```diff
@@ -36,6 +37,29 @@ pub struct Message {
...
+    pub reply: Option<String>,
+
+    /// The message contents.
+    pub data: Vec<u8>,
+
+    /// Optional headers associated with this `Message`.
+    pub headers: Option<Headers>,
 
     /// Client for publishing on the reply subject.
     #[doc(hidden)]
```
Comment (caspervonb) [inline src/message.rs:65]: Since client is hidden from doc, maybe we should just make this private?

**Review by caspervonb — APPROVED** [review]
LGTM aside from lints

**Pull request merged**